File: IM.eSuite.Web/src/app/modules/organization/models/plant.ts
import { SelectItem } from "@im-angular/core";
import { Organization } from "./organization";
import { Country } from "./country";
import { SOA } from "./soa";

export class Plant extends Organization {    
    public gaiaCode: number;
    public countryId: number;
    public country: Country;
    public soaId: number;
    public soa: SOA;

    constructor(plant?: Plant){
        super(plant);
        if(plant){
            this.gaiaCode = plant.gaiaCode;
            this.countryId = plant.countryId;
            this.soaId = plant.soaId;
        }

        this.country = new Country();
        this.soa = new SOA();
    }

    public static toSelectItem(plant: Plant): SelectItem {
        let result = new SelectItem();
        result.id = plant.id;
        result.text = `${plant.gaiaCode} - ${plant.name}`;
        return result;
    }

    public static comparePlantByName(p1: Plant, p2: Plant): number {
        if(p1.name > p2.name){
            return 1;
        }

        if(p1.name < p2.name){
            return -1;
        }

        return 0;
    }
}

File: IM.eSuite.Domain/Model/Plant.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using IM.NETCore.Domain;

namespace IM.eSuite.Domain
{
    public class Plant : EntityBase
    {
        /// <summary>
        /// Name of the plant.
        /// </summary>
        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        /// <summary>
        /// Gaia code of the plant.
        /// </summary>
        [Required]
        [MinLength(6)]
        [MaxLength(6)]
        public string GaiaCode { get; set; }

        /// <summary>
        /// Country id of the plant.
        /// </summary>
        public int CountryId { get; set; }

        /// <summary>
        /// SOA id of the plant.
        /// </summary>
        public int SOAId { get; set; }

        /// <summary>
        /// Flag indicating wether plant is deleted or not.
        /// </summary>
        public bool IsDeleted { get; set; }

        /// <summary>
        /// Plant's country
        /// </summary>
        public virtual Country Country { get; set; }

        /// <summary>
        /// Plant's SOA
        /// </summary>
        public virtual SOA SOA { get; set; }

        /// <summary>
        /// Plant's departments
        /// </summary>
        public virtual ICollection<Organization> Departments { get; set; }

        /// <summary>
        /// Plant's users
        /// </summary>
        public virtual ICollection<UserPlant> Users { get; set; }
    }
}